models/members.go

```diff
@@ -53,26 +55,78 @@ var MemberAPI MemberInterface = new(memberAPI)
 type MemberInterface interface {
 	DeleteMember(id string) error
 	GetMember(id string) (Member, error)
-	GetMembers(maxResult uint8, page uint16, sortMethod string) ([]Member, error)
+	// GetMembers(maxResult uint8, page uint16, sortMethod string) ([]Member, error)
+	GetMembers(req MemberArgs) ([]Member, error)
 	InsertMember(m Member) error
 	SetMultipleActive(ids []string, active int) error
 	UpdateMember(m Member) error
 }
 
-func (a *memberAPI) GetMembers(maxResult uint8, page uint16, sortMethod string) ([]Member, error) {
-	var (
-		result []Member
-		err    error
-	)
-	query := fmt.Sprintf(`SELECT * FROM members where active != %d ORDER BY %s LIMIT ? OFFSET ?`, int(MemberStatus["delete"].(float64)), orderByHelper(sortMethod))
-	// query, _ := generateSQLStmt("get_all", "members", sortString)
+// MemberArgs is used to hold url/payload parameters while querys
+// NullBool types should be put last, so that default type like string,
+// could be parsed during form parsing
+type MemberArgs struct {
+	BasicArgs
+	Active       string   `form:"active" db:"active"`
+	CustomEditor NullBool `form:"custom_editor" db:"custom_editor"`
+}
+
+func (args *MemberArgs) parse() (restricts string, values []interface{}) {
+	input := reflect.ValueOf(args).Elem()
+	whereString := make([]string, 0)
+	for i := 0; i < input.NumField(); i++ {
+		tag := input.Type().Field(i).Tag.Get("form")
+		switch tag {
+		case "custom_editor":
+			if args.CustomEditor.Valid {
+				whereString = append(whereString, "custom_editor = ?")
+				values = append(values, args.CustomEditor.Bool)
+			}
+		case "active":
+			if args.Active != "" {
+				tmp := map[string][]int{}
+				err := json.Unmarshal([]byte(args.Active), &tmp)
+				if err != nil {
+					fmt.Println("active ", err.Error())
+				} else {
+					for k, v := range tmp {
+						whereString = append(whereString, fmt.Sprintf("%s %s (?)", tag, operatorHelper(k)))
+						values = append(values, v)
+					}
+				}
+			}
```

Comment:
I think we should have the default to handle the exception here.

Reply:
I don't understand what you mean by default. Do you mean active should be set default to not -1 ?